operators/src/source/gdal_source.rs

```diff
@@ -931,11 +938,13 @@ mod tests {
 
         assert_eq!(c.len(), 4);
 
+        let tile_1 = &c[0];
+
         assert_eq!(
-            c[0].time,
+            tile_1.time,
             TimeInterval::new_unchecked(1_385_856_000_000, 1_388_534_400_000)
         );
 
-        assert!(!c[0].grid_array.data.iter().any(|p| *p != 0)); // TODO: use actual no data value
+        assert!(!tile_1.grid_array.data.iter().any(|p| *p != 0)); // TODO: the NDVI data has no NO DATA value. Currently, we fill empty areas with 0 if there is no NO DATA. We have to add a strategy to fix this since we meight add empty areas through the tiling approach.
```

Comment:
with our tiling approach we will actually almost certainly have no data values because the outer tile have always pixels outside of the valid CRS bounds. I think we need to enforce a no data value for all data sets. This might mean to use the next bigger data type if all the values are used up by the measurement.

Also: typo: "meight" -> might

Reply:
how would we do this? we need to open the dataset to check if there is a no-data-value. but to open the dataset we need the type...

Reply:
In my opinion the datasetdb should know the no-data value for every data set in the system. At some point it gets added/imported and then we can specify and store it.

Reply:
where is it imported? What kind of operator does this? Is this a geoengine2 which can handle this process?

Reply:
at some point you have to say here is a raster time series whose files lie in path x and this is how to get from a query time to a file name. That's where we can also say the no data value is always x. or the no data value is x before date y and after that it's y.

Reply:
i guess we will need to scan the files and a) use the metadata from that scan directly or b) store them in the database.